File: Dante2/Module2.07/Zadanie1.01/int_operations.c
#include "int_operations.h"

int add_int(int a, int b)
{
    return a + b;
}

int sub_int(int a, int b)
{
    return a- b;
}

int mul_int(int a, int b)
{
    return a * b;
}

int div_int(int a, int b)
{
    if(b == 0)
    {
        return 0;
    }
    return a / b;
}

int calculate(int a, int b, int (*type)(int, int))
{
    return type(a, b);
}

File: Dante2/Module2.07/Zadanie1.01/int_operations.rs
pub fn add_int(a: i32, b: i32) -> i32 {
    a + b
}
pub fn sub_int(a: i32, b: i32) -> i32 {
    a - b
}
pub fn mul_int(a: i32, b: i32) -> i32 {
    a * b
}
pub fn div_int(a: i32, b: i32) -> i32 {
    if b == 0 {
        return 0;
    }
    a / b
}

pub fn calculate(a: i32, b: i32, f: fn(i32, i32) -> i32) -> i32 {
    f(a, b)
}
